src/deployconfig.jl

```diff
@@ -484,6 +484,52 @@ function post_github_status(type::S, deploydocs_repo::S, sha::S, subfolder=nothi
 end
 
 
+"""
+    SimpleSSHConfig <: DeployConfig
+
+Implementation of deploy config for deploying directly via SSH.  This is particularly useful in cases
+in which there is a private repository and it's impractical or impossible to set up GitHub Actions or
+Travis for fully automated deployment.
+
+The constructor optionally accepts a path to the SSH key, so that it is possible to use multiple distinct keys.
+If the key location is not passed or is an empty string, it will check for the `DOCUMENTER_KEY_PATH` environment
+variable for a path to an SSH key.  If `DOCUMENTER_KEY_PATH` is not set, it will check for `DOCUMENTER_KEY`
+as in the Travis and GitHub Actions configs.  Failing this, it will check for a key at `\$HOME/.ssh/id_rsa`.
+"""
```

Comment:
Happy to leave the tagging as a future enhancement, but I am slightly concerned by someone just doing `git checkout vX.Y.Z` and running `make.jl`, expecting it to deploy the tag. So I think it would be good to have a warning for this in the docs:

```suggestion

!!! warning "Tagged versions"

    `SimpleSSHConfig` does not currently support pushing documentation for tagged versions, and
    will always deploy the documentation into development version directory in the remote
    repository.
"""
```